devdox/app/repositories/repo.py

```
@@ -49,3 +71,11 @@ async def create_new_repo(self, repo_model: Repo) -> Repo:
         await repo_model.save(force_create=True)
         await repo_model.refresh_from_db()
         return repo_model
+
+    async def get_by_id(self, repo_id: str) -> Repo:
+        try:
+            return await Repo.get(id=repo_id)  # Raises DoesNotExist if not found
+        except DoesNotExist:
+            raise internal_error(
+                **self.InternalExceptions.REPOSITORY_DOESNT_EXIST.value
+            )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)